docs/sources/administration/cli.md

```diff
@@ -10,41 +10,217 @@ weight = 8
+### Display Grafana CLI version
 
-then there are two flags that can be used to set homepath and the config file path.
+`--version` or `-v` prints the version of Grafana CLI currently running.
 
-`grafana-cli --homepath "/usr/share/grafana" admin reset-admin-password newpass`
+**Example:**
+```bash
+grafana-cli -v
+```
 
-If you have not lost the admin password then it is better to set in the Grafana UI. If you need to set the password in a script then the [Grafana API](http://docs.grafana.org/http_api/user/#change-password) can be used. Here is an example using curl with basic auth:
+### Override default plugin directory
 
+`--pluginsDir value` overrides the path to where your local Grafana instance stores plugins. Use this option if you want to install, update, or remove a plugin somewhere other than the default directory ("/var/lib/grafana/plugins") [$GF_PLUGIN_DIR].
+
+**Example:**
 ```bash
-curl -X PUT -H "Content-Type: application/json" -d '{
-  "oldPassword": "admin",
-  "newPassword": "newpass",
-  "confirmNew": "newpass"
-}' http://admin:admin@<your_grafana_host>:3000/api/user/password
+grafana-cli --pluginsDir "/var/lib/grafana/devplugins" plugins install <plugin-id>
 ```
+
+### Override default plugin repo URL
+
+`--repo value` allows you to download and install or update plugins from a repository other than the default Grafana repo.
+
+**Example:**
+```bash
+grafana-cli --repo "https://example.com/plugins" plugins install <plugin-id>
+```
+
+### Override default plugin .zip URL
+
+`--pluginUrl value` allows you to download a .zip file containing a plugin from a local URL instead of downloading it from the default Grafana source.
+
+**Example:**
+```bash
+grafana-cli --pluginUrl https://company.com/grafana/plugins/<plugin-id>-<plugin-version>.zip plugins install <plugin-id>
+```
+
+### Override Transport Layer Security
+
+`--insecure` allows you to turn off Transport Layer Security (TLS) verification (insecure). You might want to do this if you are downloading a plugin from a non-default source.
+
+**Example:**
+```bash
+grafana-cli --insecure --pluginUrl https://company.com/grafana/plugins/<plugin-id>-<plugin-version>.zip plugins install <plugin-id>
+```
+
+### Enable debug logging
+
+`--debug` or `-d` enables debug logging.
+> Where does is the log stored? Why would you want to use this? Is my example in any way realistic?      
+
+**Example:**
+```bash
+grafana-cli --debug plugins install <plugin-id>
+```
+
+### --configOverrides value  
+> In plain terms, what does this mean?
+Configuration options to override defaults as a string. For example: cfg:default.paths.log=/dev/null
+
+**Example:**
+```bash
+grafana-cli --configOverrides cfg:default.paths.log=/dev/null plugins install <plugin-id>
+```
+
+### --homepath value         
+> In plain terms, what does this mean?
+Sets the path for the Grafana install/home path, defaults to working directory.
+> Why would you want to use this? Is my example in any way realistic?  
```

Comment:
You would not want this for installing plugins. The plugin commands are manipulating files (downloading and extracting them into the plugins directory or deleteing files from the plugins directory). 

This is a dark and murky part of Grafana. The `homepath` flag is needed when doing something with the database - like resetting the admin password. This is a technical implementation detail that has been exposed and makes life difficult for the user. The CLI needs to know where Grafana is installed to get the config file, which is needed to get the database login details.

This will work without the `homepath` flag if you are in the Grafana directory when using the CLI.

See here for examples of the problems this has caused: https://community.grafana.com/t/admin-password-reset/19455/9 and https://github.com/grafana/grafana/issues/12638

Reply:
Updated.